### PR TITLE
Fix builds

### PR DESCRIPTION
**Pull request by samuel-massinon**
https://github.com/invenia/FTPClient.jl/pull/21 Should be merged first, this is just so I can build with appveyor.

### CODE REVIEW

**Comment by samuel-massinon** [comment]
@omus Ready to review.

File: src/FTPObject.jl [hunk 1]
```diff
@@ -104,7 +104,9 @@ Upload IO object "file" to the FTP server and save as "file_name".
 If "file" is not specified, the file "file_name" is uploaded.
 """ ->
 function upload(ftp::FTP, file_name::AbstractString, file=nothing)
```
Comment (omus) [inline src/FTPObject.jl]: This is a very strange function signature. Maybe it should be broken up into multiple methods?

``` julia
upload(ftp::FTP, file_name::AbstractString)  # Upload a file named "file_name"
upload(ftp::FTP, file::IO)  # Upload the contents of "file" using the name associated with the IO object
upload(ftp::FTP, file::IO, file_name::AbstractString)  # Upload the contents of "file" to the file named "file_name"
```

File: test/test_client_errors.jl [hunk 2]
```diff
@@ -69,8 +69,10 @@ end
 
 context("FTP object error when uploading, non-blocking") do
     ftp = FTP(ssl=false, user=user, pswd=pswd, host=host)
-    ref = non_block_upload(ftp, upload_file)
+    file = open(upload_file)
+    ref = non_block_upload(ftp, upload_file, file)
     @fact_throws RemoteException get_upload_resp(ref)
+    close(file)
```
Comment (omus) [inline test/test_client_errors.jl]: Ideally you should use the `open(f::Function, args...)` call and with `do`:

``` julia
open(upload_file) do file
    ref = non_block_upload(ftp, upload_file, file)
    @fact_throws RemoteException get_upload_resp(ref)
end
```

File: test/test_ftp_object.jl [hunk 3]
```diff
@@ -104,8 +104,10 @@ buff = get_download_resp(ref)
 println("\nTest 37 passed.\n$(ftp)")
 
 # test 38, upload a file using blocking function
-ref = non_block_upload(ftp, upload_file)
+file = open(upload_file)
+ref = non_block_upload(ftp, upload_file, file)
 get_upload_resp(ref)
+close(file)
```
Comment (omus) [inline test/test_ftp_object.jl]: Another good candidate for a open/do.

File: src/FTPObject.jl (hunk 1 above)
Comment (samuel-massinon) [inline src/FTPObject.jl]: There might be a need to upload something with a different name. Maybe add `upload_name::AbstractString=nothing` to each function too.

Comment (omus) [inline src/FTPObject.jl]: If my memory serves me correctly FTP clients usually name things in terms of "remote" and "local". Probably the `upload` function would have the following methods:

``` julia
upload(ftp::FTP, local_name::AbstractString)
upload(ftp::FTP, local_name::AbstractString, remote_name::AbstractString)
upload(ftp::FTP, local_file::IO)
upload(ftp::FTP, local_file::IO, remote_name::AbstractString)
```

Comment (samuel-massinon) [inline src/FTPObject.jl]: ah yes, I like that.

File: README.md [hunk 4]
```diff
@@ -84,7 +84,9 @@ ftp_close_connection(ctxt::ConnContext)
 FTP(;host="", block=true, implt=false, ssl=false, ver_peer=true, act_mode=false, user="", pswd="", binary_mode=true)
 close(ftp::FTP)
 download(ftp::FTP, file_name::String, save_path::String="")
-upload(ftp::FTP, file_name::String, file=nothing)
+upload(ftp::FTP, local_name::AbstractString)
+upload(ftp::FTP, local_name::AbstractString, remote_name::AbstractString)
```
Comment (omus) [inline README.md:88]: If you were using the syntax in the Julia manual you could do `upload(ftp, local_name, [remote_name])`

**Pull request merged**